darmanis_asap_data_prep.Rmd

```diff
@@ -0,0 +1,109 @@
+---
+title: "ASAP_data_prep"
+author: "C. Savonen, CCDL for ALSF"
+date: "12/6/2018"
+output: html_document
+---
+
+## Purpose: Prep data for use in [ASAP online](https://asap.epfl.ch/)
+```{r Import functions}
+source(file.path("scripts", "asap_prep_functions.R"))
+```
+
+```{r Import data}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Import gene expression matrix data
+data <- readr::read_tsv(file.path("darmanis_data", "salmon_quants",
+                                  "darmanis_data_counts.tsv"))
+# Store gene names separately
+gene <- data$gene
+```
+
+### Import metadata
+```{r Set up metadata}
+if (file.exists(file.path("darmanis_data", "meta_data.csv"))) {
+# Read in csv if it's been made, if not read in from GEOquery
+  geo.meta <- read.csv(file.path("darmanis_data", "meta_data.csv"), 
+                       stringsAsFactors = FALSE)
+} else {
+  geo.meta <- GEOquery::getGEO("GSE84465", destdir = "darmanis_data")
+  geo.meta <- data.frame(geo.meta[[1]]@phenoData@data)
+}
+```
+
+### Make sure the samples and the metadata are in the same order.
+```{r Order samples}
+# Put samples in same order 
+id.key <- readRDS(file.path("darmanis_data", "sample_id_key.RDS"))
+
+# Make a sample conversion key
+convert.key <- as.list(as.character(id.key$gsm.ids))
+names(convert.key) <- as.character(id.key$run)
+
+# Obtain GSM ids using conversion key and make these the column names 
+colnames(data) <- dplyr::recode(colnames(data), !!!convert.key)
+
+# Only keep metadata for samples in the set
+geo.meta <- geo.meta[match(colnames(data)[-1], geo.meta$geo_accession), ]
+```
+
+### Make batch file for ASAP to correct batches by.
+In this case we will use the plate ids. 
+ASAP wants the first column to be the sample IDs that match your sample data file
+and the second column to be the corresponding batch the sample belongs to.
+```{r Make batch info}
+# Make batch info matrix
+batch.info <- cbind(geo.meta$geo_accession, geo.meta$plate.id.ch1)
+
+# Let's exclude batches of "1"
+batch.sum <- summary(as.factor(batch.info[,2]))
+batch.one <- names(batch.sum)[which(batch.sum <= 2)]
```

Comment:
Why is this value `2`?

Reply:
The batches of 2 are apparently too small for KNN evaluations so I went back to exclude these samples as well. These small batches are from a couple things. 1) Not all the darmanis data is done processing (I'm running more samples right now). 2) After I exclude low percentage mapped read samples, some of the batches get smaller. What I need to do after I have the complete dataset, is determine if there are entire batches that should be thrown out. ie, if 95 out of 96 of samples of a particular batch have very low mapped reads, probably should just get rid of #96 as well. I haven't been able to do this analysis because not all ~3500 samples are done yet.

Reply:
However, as you noted on the other PR, the KNN accuracy statistics are not really appropriate for this imbalanced batch groupings, so I'm gonna go back and try kappa instead. Still, the batches of 2 might still need to be excluded. IDK what is a good number of samples per batch to keep in.

Reply:
When I looked at the comment and the variable name at the time of review, I would expect you to be selecting everything greater than 1. If comments are out of date when they get to me and I'm trying to verify correctness, that makes getting things through review more challenging. The updated version looks good 👍